### PR TITLE
kernel: bump 6.6 to 6.6.25

### PR DESCRIPTION
**Pull request by graysky2**
kernel: bump 6.6 to 6.6.25
    
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.25
    
All patches automatically rebased.
    
Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000

### CODE REVIEW

**Comment by graysky2** [comment]
@Ansuel Please check my manual rebase of your patch.  I changed:
```
config.add_legacy_fixed_of_cells = of_device_is_compatible(node, "nvmem-cells");
```
to
```
config.add_legacy_fixed_of_cells = !of_node_name_eq(mtd->dev.of_node, "nand");
```

**Comment by zxlhhyccc** [comment]
@graysky2 Is kernel 6.1 no longer updated?

File: target/linux/ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch [hunk 1]
```diff
@@ -47,12 +47,12 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
 
 --- a/drivers/mtd/mtdcore.c
 +++ b/drivers/mtd/mtdcore.c
-@@ -933,7 +933,7 @@ static struct nvmem_device *mtd_otp_nvme
- 	config.name = compatible;
- 	config.id = NVMEM_DEVID_AUTO;
```
Comment (DragonBluep) [inline target/linux/ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch:52]: I think this patch can be removed as it's the same fix as https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807.

**Comment by namiltd** [comment]
PR https://github.com/openwrt/openwrt/pull/14868 is merged. Please check if your patch requires an update

File: target/linux/ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch (hunk 1 above)
Comment (graysky2) [reply target/linux/ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch:52]: Thanks, fixed.

Comment (namiltd) [reply target/linux/ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch:52]: My proposal combining the fix https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807 and https://github.com/openwrt/openwrt/pull/14940/commits/8ec76705d42766156c4185f59998c7144f815363
```
-	config.add_legacy_fixed_of_cells = !!np;
+	config.add_legacy_fixed_of_cells = !!np && !of_node_name_eq(mtd->dev.of_node, "nand");

Comment (namiltd) [reply target/linux/ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch:52]: My proposal combining the fix https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807 and https://github.com/openwrt/openwrt/pull/14940/commits/8ec76705d42766156c4185f59998c7144f815363
```
-	config.add_legacy_fixed_of_cells = !!np;
+	config.add_legacy_fixed_of_cells = !!np && !of_node_name_eq(mtd->dev.of_node, "nand");

**Review by namiltd — COMMENTED** [review]
My proposal combining the fix https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807 and https://github.com/openwrt/openwrt/pull/14940/commits/8ec76705d42766156c4185f59998c7144f815363

**Comment by namiltd** [comment]
PR https://github.com/openwrt/openwrt/pull/1477 merged, Please check if your patch requires an update.

**Pull request merged**